### PR TITLE
MAYA-125275 preserve the layer manager expanded layers

### PR DESCRIPTION
**Pull request by pierrebai-adsk**
When the layer manager model is rebuilt, preserve the state of the layer view (currently, only the expanded state of each layer) so that it can be restored once the model rebuilt is done.

- Add a LayerViewMemento to preserve the view state.
- Provide a preserve() and restore() method to save and restorethe state.
- Provide an empty method to avoid losing state when multiple calls to About-to-reset are made.
- Preserve befor ethe model reset, restore after.
- Note that, yes Qt emits the signal multiple times when the model is reset, so we have to take that into accouunt.

### CODE REVIEW

File: lib/usd/ui/layerEditor/layerTreeView.cpp [hunk 1]
```diff
@@ -131,7 +136,77 @@ void LayerTreeView::onItemDoubleClicked(const QModelIndex& index)
...
+        PXR_NS::SdfLayerRefPtr layer = item->layer();
+        if (layer) {
+            const ItemId id = layer->GetIdentifier();
+            const auto   state = itemsState.find(id);
+            if (state != itemsState.end()) {
+                expanded = state->second.expanded;
+            }
+        }
+
+        view.setExpanded(item->index(), expanded);
```
Comment (seando-adsk) [inline lib/usd/ui/layerEditor/layerTreeView.cpp:184]: When you call things like this you trigger a redraw. So normally when doing this in a loop, we'll first disable drawing updates before the loop, and then enable them (and redraw) after the loop.

**Review by seando-adsk — CHANGES_REQUESTED** [review]
One minor variable renaming and a question about drawing.

File: lib/usd/ui/layerEditor/layerTreeView.cpp (hunk 1 above)
Comment (pierrebai-adsk) [reply lib/usd/ui/layerEditor/layerTreeView.cpp:184]: Do we have a utility class that disable Qt redraws and trigger and repaint already existing?

File: lib/usd/ui/layerEditor/qtUtils.h [hunk 2]
```diff
@@ -53,6 +53,20 @@ class QtUtils
     }
 };
 
+/**
+ * @brief Disable repaint updates for the given widget until the disabler is destoryed.
```
Comment (seando-adsk) [inline lib/usd/ui/layerEditor/qtUtils.h:57]: Minor typo: destoryed -> destroyed

File: lib/usd/ui/layerEditor/layerTreeView.h [hunk 3]
```diff
@@ -31,11 +31,37 @@ namespace UsdLayerEditor {
...
+    bool empty() const { return itemsState.empty(); }
+
+private:
+    using ItemId = std::string;
+    struct ItemState
+    {
+        bool _expanded = false;
+    };
+
+    std::map<ItemId, ItemState> itemsState;
```
Comment (seando-adsk) [inline lib/usd/ui/layerEditor/layerTreeView.h:62]: Sorry this one also needs an _

**Review by seando-adsk — CHANGES_REQUESTED** [review]
Still one member variable to rename and one typo.

**Review by seando-adsk — APPROVED** [review]

**Pull request merged**